File: software_challenge_sdk/src/server_api/sc/protocol/requests/ControlTimeoutRequest.java
package sc.protocol.requests;

import com.thoughtworks.xstream.annotations.XStreamAlias;
import com.thoughtworks.xstream.annotations.XStreamAsAttribute;
import sc.protocol.responses.ProtocolMessage;

@XStreamAlias("timeout")
public class ControlTimeoutRequest  extends ProtocolMessage implements ILobbyRequest{
  @XStreamAsAttribute
  public String roomId;

  @XStreamAsAttribute
  public boolean activate;

  @XStreamAsAttribute
  public int slot;

  public ControlTimeoutRequest(String roomId, boolean activate, int slot) {
    this.roomId = roomId;
    this.activate = activate;
    this.slot = slot;
  }

}

File: socha-sdk/src/server-api/sc/protocol/requests/ControlTimeoutRequest.kt
package sc.protocol.requests

import com.thoughtworks.xstream.annotations.XStreamAlias
import com.thoughtworks.xstream.annotations.XStreamAsAttribute

/** Used to change whether a player in a slot can time out. */
@XStreamAlias("timeout")
data class ControlTimeoutRequest(
        @XStreamAsAttribute
        val roomId: String,
        @XStreamAsAttribute
        val activate: Boolean,
        @XStreamAsAttribute
        val slot: Int
): AdminLobbyRequest
